pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -107,8 +107,31 @@ class GraphAgentDeps(Generic[DepsT, ResultDataT]):
     run_span: logfire_api.LogfireSpan
 
 
+class AgentNode(BaseNode[GraphAgentState, GraphAgentDeps[DepsT, Any], result.FinalResult[NodeRunEndT]]):
```

Comment:
I tried using a type-alias instead of a class here, and it caused some issues because I couldn't inherit from it.

I think may be possible to drop the `AgentNode` class and use a type alias (just inheriting from the non-aliased parametrized BaseNode), and either way I think it's not currently public, but I think having this type allows us to remove most references to `BaseNode` in `agent.py`, which I think ultimately makes a lot of types more readable.

Unless one of you really objects, I would prefer to keep it around for now rather than wrestle with the consequences of removing it (in terms of verbosity and/or type-checking challenges). Especially considering it isn't a public API anyway.